internal/api/metadata/describe.go

```diff
@@ -14,7 +17,35 @@ type DescribeResponse struct {
 
 // Describe all endpoints on the server.
 func HandleDescribe(request *DescribeRequest) (*DescribeResponse, *core.APIError) {
-	response := DescribeResponse{core.GetAPIDescription()}
+	response := DescribeResponse{}
+
+	if !request.ForceCompute {
+		apiDescription, err := core.GetAPIDescription()
+		if err != nil {
+			log.Warn("Unable to get cached API description.", err)
+		}
+
+		if apiDescription != nil {
+			response.APIDescription = *apiDescription
+			return &response, nil
+		}
+	}
+
+	routes := core.GetAPIRoutes()
+	if routes == nil || len(*routes) == 0 {
+		return nil, core.NewInternalError("-501", request, "Unable to describe API endpoints when the cached routes are empty.")
+	}
+
+	apiDescription, err := core.DescribeRoutes(*routes)
+	if err != nil {
+		return nil, core.NewInternalError("-502", request, "Failed to describe API endpoints.").Err(err)
+	}
+
+	core.SetAPIDescription(apiDescription)
+
+	if apiDescription != nil {
+		response.APIDescription = *apiDescription
```

Comment:
Is it possible to embed a pointer instead so you don't have to make a copy?
(IDK if it is possible.)